### ISSUE TITLE
unable to decode v3.0 protocol - timestamp incompatibility

### ISSUE THREAD

**Issue by danrubel** [dart-lang/vm_service_drivers#51]
Now that Dart SDK 1.13 dev builds are shipping, we are seeing v3.0 protocol where as Dart SDK 1.12 stable is v2.x. When the current client is run with 1.13, the following exception occurs. Apparently
timestamps in v2.x were int while in v3.0 they are double as the exception below shows.

```
Observatory listening on http://127.0.0.1:53778
Using sdk at /Users/danrubel/work/eclipse3/dart-sdk.
dart process started
Observatory listening on http://127.0.0.1:7581

socket connected
--> {"id":"1","method":"streamListen","params":{"streamId":"Isolate"}}
--> {"id":"2","method":"streamListen","params":{"streamId":"Debug"}}
--> {"id":"3","method":"streamListen","params":{"streamId":"Stdout"}}
--> {"id":"4","method":"getVM","params":{}}
<-- {"jsonrpc":"2.0","id":"1","result":{"type":"Success"}}
<-- {"jsonrpc":"2.0","id":"2","result":{"type":"Success"}}
<-- {"jsonrpc":"2.0","id":"3","result":{"type":"Success"}}
unable to decode message: {"jsonrpc":"2.0", "result":{"type":"VM","architectureBits":32,"targetCPU":"ia32","hostCPU":"Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz","version":"1.13.0-edge.7b322c246c4bd3670294485c70dad0ab87adb7de (Wed Sep 16 07:21:34 2015) on \"macos_ia32\"","pid":"24537","_assertsEnabled":false,"_typeChecksEnabled":false,"startTime":1442638868575.000000,"isolates":[{"type":"@Isolate","fixedId":true,"id":"isolates\/535398960","name":"sample_main.dart$main","number":"535398960"}]},"id":"4"}, type 'double' is not a subtype of type 'int' of 'value'.
#0      VM.startTime= (package:vm_service_lib/vm_service_lib.dart:1779:7)
#1      VM.VM.fromJson (package:vm_service_lib/vm_service_lib.dart:1758:5)
#2      VM.parse (package:vm_service_lib/vm_service_lib.dart:1749:36)
#3      createObject (package:vm_service_lib/vm_service_lib.dart:291:34)
#4      VmService._processMessage (package:vm_service_lib/vm_service_lib.dart:269:32)
#5      _RootZone.runUnaryGuarded (dart:async/zone.dart:1103)
#6      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341)
#7      _DelayedData.perform (dart:async/stream_impl.dart:595)
#8      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:711)
#9      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:671)
#10     _microtaskLoop (dart:async/schedule_microtask.dart:43)
#11     _microtaskLoopEntry (dart:async/schedule_microtask.dart:52)
#12     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:96)
#13     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:151)

<-- {"jsonrpc":"2.0", "result":{"type":"VM","architectureBits":32,"targetCPU":"ia32","hostCPU":"Intel(R) Core(TM) i7-4750HQ CPU @ 2.00GHz","version":"1.13.0-edge.7b322c246c4bd3670294485c70dad0ab87adb7de (Wed Sep 16 07:21:34 2015) on \"macos_ia32\"","pid":"24537","_assertsEnabled":false,"_typeChecksEnabled":false,"startTime":1442638868575.000000,"isolates":[{"type":"@Isolate","fixedId":true,"id":"isolates\/535398960","name":"sample_main.dart$main","number":"535398960"}]},"id":"4"}
```

**Comment by devoncarew** [comment]
@danrubel, I think this is done as well?

**Comment by danrubel** [comment]
Yes, this is fixed in latest 1.13 build.
